### PR TITLE
Ato 1016/create custom domain name for ipv stub

### PR DESCRIPTION
**Pull request by isaac-GDS**
Create a custom domain for the ipv orchestration stub

Create hosted zone on Orch account
Create associated NS record in root domain account via manual deployment
Create Certificate, Gateway domain , gateway mapping and A record via manual deployment

### CODE REVIEW

File: infrastructure/manual-stacks/link-hosted-zone/template.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Resources:
+  IpvStubNSRecord:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Type: NS
+      Name: !FindInMap [ EnvironmentConfiguration, !Ref Environment, ipvStubDomainName ]
+      HostedZoneId: !FindInMap [ EnvironmentConfiguration, !Ref Environment, oidcHostedZoneId ]
+      ResourceRecords: !FindInMap [ EnvironmentConfiguration, !Ref Environment, ipvStubResourceRecords ]
+      TTL: 172800
```
Comment (isaac-GDS) [inline infrastructure/manual-stacks/link-hosted-zone/template.yaml:48]: Does this value need to be so high ?

File: infrastructure/manual-stacks/link-hosted-zone/template.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  EnvironmentConfiguration:
+    dev:
+      oidcHostedZoneId: Z04812572XHS4OV4NR9AM
+      ipvStubDomainName: ipvStub.oidc.sandpit.account.gov.uk
+      ipvStubResourceRecords: [
+            ns-1660.awsdns-15.co.uk,
+            ns-550.awsdns-04.net,
+            ns-477.awsdns-59.com,
+            ns-1188.awsdns-20.org
+          ]
```
Comment (isaac-GDS) [inline infrastructure/manual-stacks/link-hosted-zone/template.yaml:26]: The equivalent will be added to other envs when this PR is merged and deployed.

File: ipv-stub-template.yml [hunk 3]
```diff
@@ -21,6 +21,19 @@ Parameters:
...
+Mappings:
+  EnvironmentConfiguration:
+    dev:
+      ipvStubDomainName: ipvStub.oidc.sandpit.account.gov.uk
+    build:
+      ipvStubDomainName: ipvStub.oidc.build.account.gov.uk
+    staging:
+      ipvStubDomainName: ipvStub.oidc.staging.account.gov.uk
+    integration:
+      ipvStubDomainName: ipvStub.oidc.build.account.gov.uk
```
Comment (billrdunn) [inline ipv-stub-template.yml:33]: `ipvStubDomainName: ipvStub.oidc.integration.account.gov.uk`

File: ipv-stub-template.yml [hunk 4]
```diff
@@ -21,6 +21,19 @@ Parameters:
     Description: The ARN of the permissions boundary to apply when creating IAM roles
     Default: none
 
+Mappings:
+  EnvironmentConfiguration:
+    dev:
+      ipvStubDomainName: ipvStub.oidc.sandpit.account.gov.uk
```
Comment (billrdunn) [inline ipv-stub-template.yml:27]: Two things:
- Feels a bit odd to have camel case in a domain name. Is it case sensitive?
- Variable could just be `stubDomain`?

File: infrastructure/manual-stacks/dns-zones/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Manual stacks - DNS
+## Intro
+
+The SAM template creates a Record and certificate for `ipvStub.oidc.account.gov.uk`
+or `ipvStub.oidc.<environment>.account.gov.uk` if environment is not `production`.
+Where in the case of the `dev` environment it gets mapped to `sandpit`.
```
Comment (billrdunn) [inline infrastructure/manual-stacks/dns-zones/README.md:6]:
```suggestion
The SAM template creates a DNS record and certificate for `ipvStub.oidc.<environment>.account.gov.uk` (or `ipvStub.oidc.account.gov.uk` if environment is `production`).
Note that the `dev` environment gets mapped to `sandpit`. 
```

File: infrastructure/manual-stacks/dns-zones/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Manual stacks - DNS
+## Intro
+
+The SAM template creates a Record and certificate for `ipvStub.oidc.account.gov.uk`
+or `ipvStub.oidc.<environment>.account.gov.uk` if environment is not `production`.
+Where in the case of the `dev` environment it gets mapped to `sandpit`. 
+
+This Stack is deployed manually once per environment
+as part of the DNS set up process.
```
Comment (billrdunn) [inline infrastructure/manual-stacks/dns-zones/README.md:9]:
```suggestion
This Stack is deployed manually once per environment as part of the DNS set up process. 
```

File: infrastructure/manual-stacks/dns-zones/template.yaml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+Mappings:
+  EnvironmentConfiguration:
+    dev:
+      ipvStubDomainName: ipvStub.oidc.sandpit.account.gov.uk
+    build:
+      ipvStubDomainName: ipvStub.oidc.build.account.gov.uk
+    staging:
+      ipvStubDomainName: ipvStub.oidc.staging.account.gov.uk
+    integration:
+      ipvStubDomainName: ipvStub.oidc.build.account.gov.uk
```
Comment (billrdunn) [inline infrastructure/manual-stacks/dns-zones/template.yaml:25]:
```suggestion
      ipvStubDomainName: ipvStub.oidc.integration.account.gov.uk
```

File: infrastructure/manual-stacks/link-hosted-zone/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# Manual stacks - Link Hosted Zone
+## Intro
+
+The SAM template is used to creates an NS Record on the root domain account for the sub domains that live
+in the orchestration account.
```
Comment (billrdunn) [inline infrastructure/manual-stacks/link-hosted-zone/README.md:5]:
```suggestion
The SAM template is used to creates an NS record on the root domain account for the subdomains that live in the orchestration account.
```

File: ipv-stub-template.yml (hunk 4 above)
Comment (isaac-GDS) [reply ipv-stub-template.yml:27]: Looked into this a bit more - route53 translates `ipvStub` just to `ipvstub` and we have that pattern elsewhere with `orchstub` and `signin` don't know if created a new hosted zone for `stub.oidc` to then create `ipv.stub.oidc..` and `auth.stub.oidc` is worth it.

File: infrastructure/manual-stacks/link-hosted-zone/template.yaml (hunk 1 above)
Comment (isaac-GDS) [reply infrastructure/manual-stacks/link-hosted-zone/template.yaml:48]: Leaving as default

**Review by billrdunn — APPROVED** [review]

**Pull request merged**